### DISCUSSION TITLE
Prevent focus on dialog

### DISCUSSION THREAD

**Discussion by kerimcet** [radix-ui/primitives#935] [Help] [answered]
I'm trying to prevent focus of first input inside dialog. I've look the docs but couldn't see 😶‍🌫️

Is there any option to fix it?

**Comment by benoitgrelard** [comment, marked as answer]
Hi @kerimcet, typically you wouldn't want to completely prevent focus inside the dialog otherwise you'll be breaking accessibility. However, if what you want is prevent the default autofocus and focus something else inside the Dialog instead, there most defeinitley is a way to do this. You can simple do `event.preventDefault()` in `Content`'s `onOpenAutoFocus` prop and then run your own focus logic in there.

![image](https://user-images.githubusercontent.com/1539897/138855270-49cb9aee-cb88-4142-825e-c03ab6a80ee4.png)

**Reply by kerimcet** [reply]
Thanks for the explanation 😊

**Reply by Sumonche016** [reply]
thanks man it works

**Reply by anthony-urbina** [reply]
thanks. this worked for me, had to use `ts-ignore` to get around type error though - using v2.1.2. Just a heads up for anyone that encounters the same.

**Reply by jiwoniverse** [reply]
In my case, `onOpenAutoFocus={(e) => e.preventDefault()}` did not work, but adding `autoFocus={false}` resolved the issue. However, using `autoFocus={false}` can degrade accessibility for users, so it should be used with caution.

**Reply by hexcowboy** [reply]
I found a solution I like better, which is making the title focusable with `tabIndex={0}` and `autoFocus`, then hiding the outline when the title is focused:

```jsx
const ModalTitle = ({
  className,
  ...props
}: React.ComponentProps<typeof Dialog.Title>) => (
  <Dialog.Title
    autoFocus
    tabIndex={0}
    className={twMerge(
      "focus:outline-none",
      className,
    )}
    {...props}
  />
);
```

**Comment by netrolite** [comment]
In my case I actually had to use the opposite - `onCloseAutoFocus`

**Comment by ellisfan** [comment]
```
const Toolbar = () => (
    <div className="relative">
        <input type="text" value={searchkeyword} />
    </div>
)
```
When using `Toolbar` within DialogContent, neither `autoFocus={false}` nor `onCloseAutoFocus={(event) => event.preventDefault()} ` works as intended.
The behavior is that the entire Dialog gains focus, displaying a blue outline on the outer layer.

**Reply by ellisfan** [reply]
ok, i must use `autoFocus` at input

**Comment by magical-chris** [comment]
In my case I just wanted to prevent a particular element from being auto-focused without preventing it from being manually focusable.

`@radix-ui/react-focus-scope` explicitly excludes anchor elements from auto-focus, so as a hack I rendered my button as:

```tsx
<a role="button" tabIndex="0" onClick={handleClick} >
```

Potentially unstable fix, but works for now. Really wish `@radix-ui/react-focus-scope` allowed opting certain elements out of autofocus via a data attribute or similar.
